### ISSUE TITLE
(anonymous namespace)::ConvertRectToGlobal(SemanticsObject*, CGRect) iOS Crash

### ISSUE THREAD

**Issue by shahriar-ax** [flutter/flutter#186005]
### Steps to reproduce

Unfortunately i can't reproduce it, it's a crash i have received from Crashlytics.

**Crash Logs:**

` Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0xc5964 __exceptionPreprocess
1  libobjc.A.dylib                0x31814 objc_exception_throw
2  Foundation                     0x906d88 -[NSMutableDictionary(NSMutableDictionary) initWithContentsOfFile:]
3  Flutter                        0x43d084 (anonymous namespace)::ConvertRectToGlobal(SemanticsObject*, CGRect) + 81 (SemanticsObject.mm:81)
4  Flutter                        0x43bff8 -[SemanticsObject globalRect] + 649 (SemanticsObject.mm:649)
5  Flutter                        0x43bf88 -[SemanticsObject accessibilityFrame] + 644 (SemanticsObject.mm:644)
6  Flutter                        0x43e83c -[FlutterScrollableSemanticsObject accessibilityBridgeDidFinishUpdate] + 176 (SemanticsObject.mm:176)
7  Flutter                        0x448a54 flutter::PlatformViewIOS::UpdateSemantics(long long, std::_fl::unordered_map<int, flutter::SemanticsNode, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::SemanticsNode>>>, std::_fl::unordered_map<int, flutter::CustomAccessibilityAction, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::CustomAccessibilityAction>>>) + 208 (accessibility_bridge.mm:208)
8  Flutter                        0x8111c8 std::_fl::__function::__func<flutter::Shell::OnEngineUpdateSemantics(long long, std::_fl::unordered_map<int, flutter::SemanticsNode, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::SemanticsNode>>>, std::_fl::unordered_map<int, flutter::CustomAccessibilityAction, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::CustomAccessibilityAction>>>)::$_0, std::_fl::allocator<flutter::Shell::OnEngineUpdateSemantics(long long, std::_fl::unordered_map<int, flutter::SemanticsNode, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::SemanticsNode>>>, std::_fl::unordered_map<int, flutter::CustomAccessibilityAction, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::CustomAccessibilityAction>>>)::$_0>, void ()>::operator()() + 1126 (__hash_table:1126)
9  Flutter                        0x80d494 flutter::Shell::OnEngineUpdateSemantics(long long, std::_fl::unordered_map<int, flutter::SemanticsNode, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::SemanticsNode>>>, std::_fl::unordered_map<int, flutter::CustomAccessibilityAction, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::CustomAccessibilityAction>>>) + 403 (function.h:403)
10 Flutter                        0x7f1e2c flutter::Engine::UpdateSemantics(long long, std::_fl::unordered_map<int, flutter::SemanticsNode, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::SemanticsNode>>>, std::_fl::unordered_map<int, flutter::CustomAccessibilityAction, std::_fl::hash<int>, std::_fl::equal_to<int>, std::_fl::allocator<std::_fl::pair<int const, flutter::CustomAccessibilityAction>>>) + 1126 (__hash_table:1126)
11 Flutter                        0x89e08c flutter::RuntimeController::UpdateSemantics(long long, flutter::SemanticsUpdate*) + 1126 (__hash_table:1126)
12 App                            0x11f30 stub CallNativeThroughSafepoint
13 App                            0xa92b24 FlutterView.__updateSemantics$Method$FfiNative (window.dart)
14 Flutter                        0x97513c dart::Simulator::DecodeSimulatorToNativeTransition(dart::Instr*) + 164 (os_thread.h:164)
15 Flutter                        0x976528 dart::Simulator::Execute(char const*, dart::Simulator::SimulatorExecutionSource) + 2198 (simulator_arm64.cc:2198)
16 Flutter                        0x8e77cc dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&) + 4261 (simulator_arm64.cc:4261)
17 Flutter                        0xa2c4d4 Dart_InvokeClosure + 4710 (dart_api_impl.cc:4710)
18 Flutter                        0x80cf5c flutter::Shell::OnAnimatorBeginFrame(fml::TimePoint, unsigned long long) + 30 (dart_invoke.cc:30)
19 Flutter                        0x7f0180 std::_fl::__function::__func<flutter::Animator::AwaitVSync()::$_0, std::_fl::allocator<flutter::Animator::AwaitVSync()::$_0>, void (std::_fl::unique_ptr<flutter::FrameTimingsRecorder, std::_fl::default_delete<flutter::FrameTimingsRecorder>>)>::operator()(std::_fl::unique_ptr<flutter::FrameTimingsRecorder, std::_fl::default_delete<flutter::FrameTimingsRecorder>>&&) + 190 (ref_ptr.h:190)
20 Flutter                        0x81f404 std::_fl::__function::__func<flutter::VsyncWaiter::FireCallback(fml::TimePoint, fml::TimePoint, bool)::$_0, std::_fl::allocator<flutter::VsyncWaiter::FireCallback(fml::TimePoint, fml::TimePoint, bool)::$_0>, void ()>::operator()() + 284 (unique_ptr.h:284)
21 Flutter                        0x46d220 fml::MessageLoopImpl::FlushTasks(fml::FlushType) + 128 (message_loop_impl.cc:128)
22 Flutter                        0x470aa4 fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*) + 86 (message_loop_darwin.mm:86)
23 CoreFoundation                 0x4dc00 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
24 CoreFoundation                 0x4d8c0 __CFRunLoopDoTimer
25 CoreFoundation                 0x4d434 __CFRunLoopDoTimers
26 CoreFoundation                 0x1dabc __CFRunLoopRun
27 CoreFoundation                 0x1ca6c _CFRunLoopRunSpecificWithOptions
28 GraphicsServices               0x1498 GSEventRunModal
29 UIKitCore                      0x9ddf8 -[UIApplication _run]
30 UIKitCore                      0x46e54 UIApplicationMain
31 UIKitCore                      0x172938 -[UIScrollView contentInset]
32 Runner                         0x7504 main (AppDelegate.swift)`

I am attaching here the whole [trace.txt](https://github.com/user-attachments/files/27371959/trace.txt).

### Expected results

The app should not crash

### Actual results

The app is crashing

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor

┌─────────────────────────────────────────────────────────┐
│ A new version of Flutter is available!                  │
│                                                         │
│ To update to the latest version, run "flutter upgrade". │
└─────────────────────────────────────────────────────────┘
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.35.1, on macOS 26.4.1 25E253 darwin-arm64, locale
    en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 36.1.0)
[✓] Xcode - develop for iOS and macOS (Xcode 26.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2025.3)
[✓] Connected device (4 available)
[✓] Network resources
```

</details>

**Comment by saurabh-mirajkar** [comment]
Thanks for the report.

Could you confirm if this is the same crash pattern as in https://github.com/flutter/flutter/issues/186003? If so, please continue the discussion there to keep things consolidated. Keeping one issue with complete context makes it easier for the team to investigate.

Kindly provide the below details:
- Verify on the latest Flutter version
- Share plugin versions (especially `connectivity_plus` if used)
- Provide any reproducible steps or patterns if observed



Marking this as `needs repro info` for now.

**Comment by shahriar-ax** [comment]
@saurabh-mirajkar 

* Current flutter version is: 3.41.9
* Current Dirt verison is: 3.11.5
* connectivity_plus: ^6.1.5
* Can't reproduced this issue

**Comment by saurabh-mirajkar** [comment]
Thanks for the confirmation. The stack trace suggests an issue with iOS semantics/accessibility, I’ve labeled this for investigation even though I cannot reproduce it locally yet.

**Comment by chunhtai** [comment]
@QuncCccccc This is just my guess, but do you think this may be related to the overlay semantics tree shape difference between hittest and traversal tree?

**Comment by QuncCccccc** [comment]
Oh I'm actually not sure. It would be better if we can have a repro sample code. I have this PR to trim traversal children in hit-test tree if the traversal parent is missing: https://github.com/flutter/flutter/pull/186826

**Comment by shahriar-ax** [comment]
> Thanks for the report.
> 
> Could you confirm if this is the same crash pattern as in [#186003](https://github.com/flutter/flutter/issues/186003)? If so, please continue the discussion there to keep things consolidated. Keeping one issue with complete context makes it easier for the team to investigate.
> 
> Kindly provide the below details:
> 
> * Verify on the latest Flutter version
> * Share plugin versions (especially `connectivity_plus` if used)
> * Provide any reproducible steps or patterns if observed
> 
> Marking this as `needs repro info` for now.

@saurabh-mirajkar 

1. using connectivity_plus: 7.0.0
2. I tried many times, but I couldn’t reproduce this issue. I guess it may be crashing only when the app is using a large amount of memory.
